module/behaviour/planning/SimpleWalkPathPlanner/src/SimpleWalkPathPlanner.cpp

```diff
@@ -207,35 +211,53 @@ namespace behaviour {
                     if (useLocalisation) {
-
                         // Transform kick target to torso space
                         Transform3D Hfw = fieldStateToTransform3D(convert<double, 3>(field.position));
                         Transform3D Htf = (Htw * Hfw.i());
-                        arma::vec3 kickTarget =
-                            Htf.transformPoint(arma::vec3({kickPlan.target[0], kickPlan.target[1], 0}));
-
-                        // //approach point:
-                        arma::vec2 ballToTarget = arma::normalise(kickTarget.rows(0, 1) - position);
-                        arma::vec2 kick_point   = position - ballToTarget * ball_approach_dist;
 
-                        if (arma::norm(position) > slowdown_distance) {
-                            position = kick_point;
-                        }
-                        else {
-                            speedFactor   = slow_approach_factor;
-                            headingChange = std::atan2(ballToTarget[1], ballToTarget[0]);
-                            sideStep      = 1;
+                        if (walkPlan) {
+                            // If walking towards kicking the ball
+                            if (walkPlan->type == WalkPlan::WalkType::BALL) {
+                                arma::vec3 kickTarget =
+                                    Htf.transformPoint(arma::vec3({kickPlan.target[0], kickPlan.target[1], 0}));
+
+                                // //approach point:
+                                arma::vec2 ballToTarget = arma::normalise(kickTarget.rows(0, 1) - position);
+                                arma::vec2 kick_point   = position - ballToTarget * ball_approach_dist;
+
+                                if (arma::norm(position) > slowdown_distance) {
+                                    position = kick_point;
+                                }
+                                else {
+                                    speedFactor   = slow_approach_factor;
+                                    headingChange = std::atan2(ballToTarget[1], ballToTarget[0]);
+                                    sideStep      = 1;
+                                }
+                            }
+                            // If walking to a point on the field
+                            else if (walkPlan->type == WalkPlan::WalkType::FIELD) {
+                                utility::math::matrix::Transform3D Hfp;
+                                Hfp.translation() = arma::vec3{walkPlan->fieldPose[0], walkPlan->fieldPose[1], 0};
+                                Hfp               = Hfp.rotateZ(walkPlan->fieldPose[2]);
+
+                                utility::math::matrix::Transform3D Htp = Htf * Hfp;
+
+                                arma::vec3 forward      = Htp.transformVector(arma::vec3{1, 0, 0});
+                                arma::vec3 goalPosition = Htp.translation();
+
+                                position      = goalPosition.rows(0, 1);
```

Comment:
only change heading when nearly there. the walk isnt stable sideways. use slowdown distance